src/custom/state/ethFlow/atoms/utils.ts

```diff
@@ -0,0 +1,56 @@
+import { atom, PrimitiveAtom, SetStateAction } from 'jotai'
+
+type Storage<Value> = {
+  getItem: (key: string) => Value
+  setItem: (key: string, newValue: Value) => void
+}
+
+const defaultStorage: Storage<unknown> = {
+  getItem: (key) => {
+    const storedValue = localStorage.getItem(key)
+    if (storedValue === null) {
+      throw new Error('no value stored')
+    }
+    return JSON.parse(storedValue)
+  },
+  setItem: (key, newValue) => {
+    localStorage.setItem(key, JSON.stringify(newValue))
+  },
+}
+
+export function atomWithStorage<Value>(
+  key: string,
+  initialValue: Value,
+  storage: Storage<Value> = defaultStorage as Storage<Value>
+): PrimitiveAtom<Value> {
+  const getInitialValue = () => {
+    try {
+      return { value: storage.getItem(key), needsSetting: false }
+    } catch {
+      return { value: initialValue, needsSetting: true }
+    }
+  }
+
+  const baseAtom = atom(initialValue)
+
+  baseAtom.onMount = (setAtom) => {
+    Promise.resolve(getInitialValue()).then(({ value, needsSetting }) => {
```

Comment:
main change from the default exported `atomWithStorage` is that this actually sets the state in localStorage if it doesn't initially exist.

Reply:
I still don't get why you need a custom version of https://jotai.org/docs/utils/atom-with-storage

You mean it doesn't initially with the initial state passed in?

Reply:
Then, why do you need to have it in the local storage if there's nothing set?
If it's not set should just use the app's default, no?